### PR TITLE
fix window function  doc string for Window length

### LINKED ISSUE #16564: DOC: signal: Many window function's input docs are wrong.
### Describe your issue.

Almost window function docs in `signal.windows` says:
https://github.com/scipy/scipy/blob/b22d64791a70975b756f69dbf2a2a2b1080394e1/scipy/signal/windows/_windows.py#L879-L881

However, if we input a negative number, a `ValueError` is thrown, not an empty array.

I think the docs should be updated and simple tests should be added.

### Reproducing Code Example

```python
In [113]: from scipy.signal.windows import barthann

In [114]: barthann(0)
Out[114]: array([], dtype=float64)

In [115]: barthann(-1)
```

### Error message

```shell
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-115-8b442405f887> in <cell line: 1>()
----> 1 barthann(-1)

/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/scipy/signal/windows/_windows.py in barthann(M, sym)
    920
    921     """
--> 922     if _len_guards(M):
    923         return np.ones(M)
    924     M, needs_trunc = _extend(M, sym)

/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/scipy/signal/windows/_windows.py in _len_guards(M)
     17     """Handle small or incorrect window lengths"""
     18     if int(M) != M or M < 0:
---> 19         raise ValueError('Window length M must be a non-negative integer')
     20     return M <= 1
     21

ValueError: Window length M must be a non-negative integer
```

### SciPy/NumPy/Python version information

1.8.1 1.23.0 sys.version_info(major=3, minor=10, micro=4, releaselevel='final', serial=0)

### PR DESCRIPTION
**Pull request by eshirvana**
#### Reference issue

closes: gh-16564

#### What does this implement/fix?
fix the document for window length in window functions

### CODE REVIEW

File: scipy/signal/windows/_windows.py [hunk 1]
```diff
@@ -128,8 +128,7 @@ def boxcar(M, sym=True):
     Parameters
     ----------
     M : int
-        Number of points in the output window. If zero or less, an empty
-        array is returned.
+        Number of points in the output window. If zero an empty array is returned.
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:131]: I think that returning an empty array if zero is provided is not so important. From a user standpoint, an exception thrown when it is negative is mote important information, it should be described.

**Review by AtsushiSakai — CHANGES_REQUESTED** [review]
Thank you for your PR. I have 2 additional comments.

1.
Could you please fix these lint errors? See "scipy.scipy (Main Lint) CI".
```
scipy/signal/windows/_windows.py:131:80: E501 line too long (82 > 79 characters)
scipy/signal/windows/_windows.py:180:80: E501 line too long (82 > 79 characters)
scipy/signal/windows/_windows.py:243:80: E501 line too long (82 > 79 characters)
scipy/signal/windows/_windows.py:306:80: E501 line too long (82 > 79 characters)
scipy/signal/windows/_windows.py:366:80: E501 line too long (82 > 79 characters)
scipy/signal/windows/_windows.py:446:80: E501 line too long (82 > 79 characters)
scipy/signal/windows/_windows.py:502:80: E501 line too long (82 > 79 characters)
scipy/signal/windows/_windows.py:548:80: E501 line too long (82 > 79 characters)
scipy/signal/windows/_windows.py:615:80: E501 line too long (82 > 79 characters)
scipy/signal/windows/_windows.py:710:80: E501 line too long (82 > 79 characters)
scipy/signal/windows/_windows.py:787:80: E501 line too long (82 > 79 characters)
scipy/signal/windows/_windows.py:869:80: E501 line too long (82 > 79 characters)
scipy/signal/windows/_windows.py:926:80: E501 line too long (82 > 79 characters)
scipy/signal/windows/_windows.py:1012:80: E501 line too long (82 > 79 characters)
scipy/signal/windows/_windows.py:1088:80: E501 line too long (82 > 79 characters)
scipy/signal/windows/_windows.py:1287:80: E501 line too long (82 > 79 characters)
scipy/signal/windows/_windows.py:1420:80: E501 line too long (82 > 79 characters)
scipy/signal/windows/_windows.py:1545:80: E501 line too long (82 > 79 characters)
scipy/signal/windows/_windows.py:1603:80: E501 line too long (82 > 79 characters)
```

2.
Could you please add a test to check whether these windows functions actually return an empty array when M=0, and throw an exception when -1?
Maybe you can add the test to this file: https://github.com/scipy/scipy/blob/main/scipy/signal/tests/test_windows.py

**Comment by eshirvana** [comment]
hi @AtsushiSakai , I see in `test_windowfunc_basics` we already have these two tests :

```
            # Check invalid lengths
            assert_raises(ValueError, window, 5.5, *params)
            assert_raises(ValueError, window, -7, *params)
```
and 
```
           # Check degenerate cases
            assert_array_equal(window(0, *params, sym=True), [])
            assert_array_equal(window(0, *params, sym=False), [])
```
aren't they already testing what we wanted to do?
otherwise, fixed the doc string and we are good to go .
please review and let me know ...
Thanks

File: scipy/signal/windows/_windows.py (hunk 1 above)
Comment (eshirvana) [reply scipy/signal/windows/_windows.py]: fixed

File: scipy/signal/windows/_windows.py [hunk 2]
```diff
@@ -1716,8 +1717,8 @@ def taylor(M, nbar=4, sll=30, norm=True, sym=True):
     Parameters
     ----------
     M : int
-        Number of points in the output window. If zero or less, an
-        empty array is returned.
+        Number of points in the output window. If zero an empty array is returned.
+        An exception thrown when it is negative.s
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:1721]: The last `s` is typo?
```suggestion
        An exception is thrown when it is negative.
```

File: scipy/signal/windows/_windows.py [hunk 3]
```diff
@@ -128,8 +128,8 @@ def boxcar(M, sym=True):
     Parameters
     ----------
     M : int
-        Number of points in the output window. If zero or less, an empty
-        array is returned.
+        Number of points in the output window. If zero an empty array
+        is returned. An exception thrown when it is negative.
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:132]:
```suggestion
        Number of points in the output window. If zero, an empty array
        is returned. An exception is thrown when it is negative.
```

**Review by AtsushiSakai — CHANGES_REQUESTED** [review]
Thanks, I missed the test suites. I have few small comments.

File: scipy/signal/windows/_windows.py [hunk 4]
```diff
@@ -178,8 +178,8 @@ def triang(M, sym=True):
...
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:182]:
```suggestion
        Number of points in the output window. If zero, an empty array
        is returned. An exception is thrown when it is negative.
```

File: scipy/signal/windows/_windows.py [hunk 5]
```diff
@@ -242,8 +242,8 @@ def parzen(M, sym=True):
...
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:246]:
```suggestion
        Number of points in the output window. If zero, an empty array
        is returned. An exception is thrown when it is negative.
```

File: scipy/signal/windows/_windows.py [hunk 6]
```diff
@@ -306,8 +306,8 @@ def bohman(M, sym=True):
...
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:310]:
```suggestion
        Number of points in the output window. If zero, an empty array
        is returned. An exception is thrown when it is negative.
```

File: scipy/signal/windows/_windows.py [hunk 7]
```diff
@@ -367,8 +367,8 @@ def blackman(M, sym=True):
...
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:371]:
```suggestion
        Number of points in the output window. If zero, an empty array
        is returned. An exception is thrown when it is negative.
```

File: scipy/signal/windows/_windows.py [hunk 8]
```diff
@@ -448,8 +448,8 @@ def nuttall(M, sym=True):
...
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:452]:
```suggestion
        Number of points in the output window. If zero, an empty array
        is returned. An exception is thrown when it is negative.
```

File: scipy/signal/windows/_windows.py [hunk 9]
```diff
@@ -505,8 +505,8 @@ def blackmanharris(M, sym=True):
...
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:509]:
```suggestion
        Number of points in the output window. If zero, an empty array
        is returned. An exception is thrown when it is negative.
```

File: scipy/signal/windows/_windows.py [hunk 10]
```diff
@@ -552,8 +552,8 @@ def flattop(M, sym=True):
...
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:556]:
```suggestion
        Number of points in the output window. If zero, an empty array
        is returned. An exception is thrown when it is negative.
```

File: scipy/signal/windows/_windows.py [hunk 11]
```diff
@@ -620,8 +620,8 @@ def bartlett(M, sym=True):
...
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:624]:
```suggestion
        Number of points in the output window. If zero, an empty array
        is returned. An exception is thrown when it is negative.
```

File: scipy/signal/windows/_windows.py [hunk 12]
```diff
@@ -716,8 +716,8 @@ def hann(M, sym=True):
...
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:720]:
```suggestion
        Number of points in the output window. If zero, an empty array
        is returned. An exception is thrown when it is negative.
```

File: scipy/signal/windows/_windows.py [hunk 13]
```diff
@@ -794,8 +794,8 @@ def tukey(M, alpha=0.5, sym=True):
...
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:798]:
```suggestion
        Number of points in the output window. If zero, an empty array
        is returned. An exception is thrown when it is negative.
```

File: scipy/signal/windows/_windows.py [hunk 14]
```diff
@@ -877,8 +877,8 @@ def barthann(M, sym=True):
...
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:881]:
```suggestion
        Number of points in the output window. If zero, an empty array
        is returned. An exception is thrown when it is negative.
```

File: scipy/signal/windows/_windows.py [hunk 15]
```diff
@@ -935,8 +935,8 @@ def general_hamming(M, alpha, sym=True):
...
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:939]:
```suggestion
        Number of points in the output window. If zero, an empty array
        is returned. An exception is thrown when it is negative.
```

File: scipy/signal/windows/_windows.py [hunk 16]
```diff
@@ -1022,8 +1022,8 @@ def hamming(M, sym=True):
...
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:1026]:
```suggestion
        Number of points in the output window. If zero, an empty array
        is returned. An exception is thrown when it is negative.
```

File: scipy/signal/windows/_windows.py [hunk 17]
```diff
@@ -1099,8 +1099,8 @@ def kaiser(M, beta, sym=True):
...
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:1103]:
```suggestion
        Number of points in the output window. If zero, an empty array
        is returned. An exception is thrown when it is negative.
```

File: scipy/signal/windows/_windows.py [hunk 18]
```diff
@@ -1215,8 +1215,9 @@ def kaiser_bessel_derived(M, beta, *, sym=True):
     Parameters
     ----------
     M : int
-        Number of points in the output window. If zero or less, an empty
-        array is returned. Note that this window is only defined for an even
+        Number of points in the output window. If zero an empty array is
+        returned.  An exception thrown when it is negative.
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:1219]:
```suggestion
        Number of points in the output window. If zero, an empty array
        is returned. An exception is thrown when it is negative.
```

File: scipy/signal/windows/_windows.py [hunk 19]
```diff
@@ -1299,8 +1300,8 @@ def gaussian(M, std, sym=True):
     Parameters
     ----------
     M : int
-        Number of points in the output window. If zero or less, an empty
-        array is returned.
+        Number of points in the output window. If zero an empty array
+        is returned. An exception thrown when it is negative.
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:1304]:
```suggestion
        Number of points in the output window. If zero, an empty array
        is returned. An exception is thrown when it is negative.
```

File: scipy/signal/windows/_windows.py [hunk 20]
```diff
@@ -1362,8 +1363,8 @@ def general_gaussian(M, p, sig, sym=True):
     Parameters
     ----------
     M : int
-        Number of points in the output window. If zero or less, an empty
-        array is returned.
+        Number of points in the output window. If zero an empty
+        array is returned. An exception thrown when it is negative.
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:1367]:
```suggestion
        Number of points in the output window. If zero, an empty array
        is returned. An exception is thrown when it is negative.
```

File: scipy/signal/windows/_windows.py [hunk 21]
```diff
@@ -1433,8 +1434,8 @@ def chebwin(M, at, sym=True):
     Parameters
     ----------
     M : int
-        Number of points in the output window. If zero or less, an empty
-        array is returned.
+        Number of points in the output window. If zero an empty array
+        is returned. An exception thrown when it is negative.
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:1438]:
```suggestion
        Number of points in the output window. If zero, an empty array
        is returned. An exception is thrown when it is negative.
```

File: scipy/signal/windows/_windows.py [hunk 22]
```diff
@@ -1559,8 +1560,8 @@ def cosine(M, sym=True):
...
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:1564]:
```suggestion
        Number of points in the output window. If zero, an empty array
        is returned. An exception is thrown when it is negative.
```

File: scipy/signal/windows/_windows.py [hunk 23]
```diff
@@ -1618,8 +1619,8 @@ def exponential(M, center=None, tau=1., sym=True):
...
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:1623]:
```suggestion
        Number of points in the output window. If zero, an empty array
        is returned. An exception is thrown when it is negative.
```

File: scipy/signal/windows/_windows.py [hunk 24]
```diff
@@ -1716,8 +1717,8 @@ def taylor(M, nbar=4, sll=30, norm=True, sym=True):
     Parameters
     ----------
     M : int
-        Number of points in the output window. If zero or less, an
-        empty array is returned.
+        Number of points in the output window. If zero an empty array is returned.
+        An exception is thrown when it is negative.
```
Comment (AtsushiSakai) [inline scipy/signal/windows/_windows.py:1721]:
```suggestion
        Number of points in the output window. If zero, an empty array
        is returned. An exception is thrown when it is negative.
```

**Review by AtsushiSakai — COMMENTED** [review]
I think all docs should be unified.

**Review by AtsushiSakai — APPROVED** [review]
LGTM. Let's hear other opinions for a few days. Then I will merge it.

**Comment by AtsushiSakai** [comment]
It looks like there are no more comments. So let's merge it. Thank you @eshirvana

**Pull request merged**